rskj-core/src/main/java/co/rsk/net/discovery/PeerExplorer.java

```diff
@@ -170,6 +177,8 @@ synchronized void handleMessage(DiscoveryEvent event) {
         //have a networkId in the message yet, then just let them through, for now.
         if (event.getMessage().getNetworkId().isPresent() &&
                 event.getMessage().getNetworkId().getAsInt() != this.networkId) {
+            logger.warn("handleMessage - Message ignored because remote peer's network id: [{}] is different from local network id: [{}]",
+                    event.getMessage().getNetworkId().getAsInt(), this.networkId);
```

Comment:
this sequence of calls is error-prone. An exception can be thrown under some circumstances, eg when networkId is not set in a message. `event.getMessage().getNetworkId()`?